Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -140,38 +140,47 @@ namespace
         jsInputSource.Set("gripSpace", Napi::External<decltype(inputSource.GripSpace)>::New(env, &inputSource.GripSpace));
 
         // Don't set hands up unless hand data is supported/available
-        if (inputSource.JointsTrackedThisFrame)
+        if (inputSource.HandTrackedThisFrame || inputSource.JointsTrackedThisFrame)
         {
-            auto handJointCollection = Napi::Array::New(env, HAND_JOINT_NAMES.size());
+            auto profiles = Napi::Array::New(env, 2);
+            profiles.Set(uint32_t{0}, Napi::String::New(env, "generic-hand-select-grasp"));
+            profiles.Set(uint32_t{1}, Napi::String::New(env, "generic-hand-select"));
+            jsInputSource.Set("profiles", profiles);
 
-            for (size_t i = 0; i < HAND_JOINT_NAMES.size(); i++)
+            if (inputSource.JointsTrackedThisFrame)
             {
-                auto napiJoint = Napi::External<std::decay_t<decltype(*inputSource.HandJoints.begin())>>::New(env, &inputSource.HandJoints[i]);
-                handJointCollection.Set(HAND_JOINT_NAMES[i], napiJoint);
-            }
+                auto handJointCollection = Napi::Array::New(env, HAND_JOINT_NAMES.size());
 
-            auto jointGetter = [handJointCollection](const Napi::CallbackInfo& info) -> Napi::Value {
-                return handJointCollection.Get(info[0].As<Napi::String>());
-            };
+                for (size_t i = 0; i < HAND_JOINT_NAMES.size(); i++)
+                {
+                    auto napiJoint = Napi::External<std::decay_t<decltype(*inputSource.HandJoints.begin())>>::New(env, &inputSource.HandJoints[i]);
+                    handJointCollection.Set(HAND_JOINT_NAMES[i], napiJoint);
+                }
 
-            handJointCollection.Set("get", Napi::Function::New(env, jointGetter, "get"));
-            handJointCollection.Set("size", static_cast<int>(HAND_JOINT_NAMES.size()));
+                auto jointGetter = [handJointCollection](const Napi::CallbackInfo& info) -> Napi::Value {
+                    return handJointCollection.Get(info[0].As<Napi::String>());
+                };
 
-            jsInputSource.Set("hand", handJointCollection);
+                handJointCollection.Set("get", Napi::Function::New(env, jointGetter, "get"));
+                handJointCollection.Set("size", static_cast<int>(HAND_JOINT_NAMES.size()));
 
-            auto profiles = Napi::Array::New(env, 2);
-            profiles.Set(uint32_t{0}, Napi::String::New(env, "generic-hand-select-grasp"));
-            profiles.Set(uint32_t{1}, Napi::String::New(env, "generic-hand-select"));
-            jsInputSource.Set("profiles", profiles);
+                jsInputSource.Set("hand", handJointCollection);
+
+            }
+            else
+            {
+                // If hand joints aren't available on a hand input, send a null hand object
+                jsInputSource.Set("hand", env.Null());
```

Comment:
>Null [](http://example.com/codeflow?start=46&length=4)

nit: should this be null or undefined?

Reply:
Null, same as when tracking is not supported
https://www.w3.org/TR/webxr-hand-input-1/#xrinputsource-interface